### PR TITLE
Add Integration test

### PR DESCRIPTION
**Pull request by benmoriceau**
## What
Add a Integration test for the s3 file transfer. It:
- Check that the file is properly deleted
- Check that the state is properly returned
- Check that the content of the file is the expected one

## Can this PR be safely reverted and rolled back?

- [X] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/task/DestinationTaskLauncher.kt [hunk 1]
```diff
@@ -139,7 +139,18 @@ class DefaultDestinationTaskLauncher(
 
     override suspend fun run() {
         exceptionHandler.setCallback { succeeded.send(false) }
-
+        log.info {
```
Comment (benmoriceau) [inline airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/task/DestinationTaskLauncher.kt:142]: This was for test

File: airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/test/util/destination_process/DockerizedDestination.kt [hunk 2]
```diff
@@ -53,6 +54,7 @@ class DockerizedDestination(
 
     private val stdoutDrained = CompletableDeferred<Unit>()
     private val stderrDrained = CompletableDeferred<Unit>()
+    // private val fileTransferMountSource = Files.createTempDirectory("tmp")
```
Comment (benmoriceau) [inline airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/test/util/destination_process/DockerizedDestination.kt:57]: It is commented for a test

File: airbyte-cdk/bulk/core/load/src/integrationTest/kotlin/io/airbyte/cdk/load/mock_integration_test/MockBasicFunctionalityIntegrationTest.kt [hunk 3]
```diff
@@ -91,4 +92,6 @@ class MockBasicFunctionalityIntegrationTest :
     override fun testBasicTypes() {
         super.testBasicTypes()
     }
+
+    @Test @Disabled override fun testBasicWriteFile() {}
```
Comment (tryangul) [inline airbyte-cdk/bulk/core/load/src/integrationTest/kotlin/io/airbyte/cdk/load/mock_integration_test/MockBasicFunctionalityIntegrationTest.kt:96]: What's this for?

Comment (tryangul) [inline airbyte-cdk/bulk/core/load/src/integrationTest/kotlin/io/airbyte/cdk/load/mock_integration_test/MockBasicFunctionalityIntegrationTest.kt:96]: Nit: Maybe leave a comment about why we are disabling.

File: airbyte-cdk/bulk/core/load/src/integrationTest/kotlin/io/airbyte/cdk/load/mock_integration_test/MockDestinationBackend.kt [hunk 4]
```diff
@@ -121,4 +121,11 @@ object MockDestinationDataDumper : DestinationDataDumper {
             MockStreamLoader.getFilename(stream.descriptor.namespace, stream.descriptor.name)
         )
     }
+
+    override fun dumpFile(
+        spec: ConfigurationSpecification,
+        stream: DestinationStream
+    ): List<String> {
+        throw NotImplementedError()
+    }
```
Comment (tryangul) [inline airbyte-cdk/bulk/core/load/src/integrationTest/kotlin/io/airbyte/cdk/load/mock_integration_test/MockDestinationBackend.kt:130]: Nit: Maybe leave a comment about why we aren't implementing.

File: airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/test/util/destination_process/DestinationProcess.kt [hunk 5]
```diff
@@ -60,6 +62,7 @@ abstract class DestinationProcessFactory {
         command: String,
         configContents: String? = null,
         catalog: ConfiguredAirbyteCatalog? = null,
+        useFileTransfer: Boolean,
```
Comment (tryangul) [inline airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/test/util/destination_process/DestinationProcess.kt:65]: Nit: I don't love that this specific feature for s3 is bleeding into the broader interface methods. Don't need to solve now, but we next time we add a specific feature, we'll need to come up with a more extensible pattern.

**Review by tryangul — APPROVED** [review]

File: airbyte-cdk/bulk/core/load/src/integrationTest/kotlin/io/airbyte/cdk/load/mock_integration_test/MockBasicFunctionalityIntegrationTest.kt (hunk 3 above)
Comment (benmoriceau) [reply airbyte-cdk/bulk/core/load/src/integrationTest/kotlin/io/airbyte/cdk/load/mock_integration_test/MockBasicFunctionalityIntegrationTest.kt:96]: I think that it is not needed for the file use case. The mocked stream loader for the file doesn't bring more value than the record one.

File: airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/test/util/destination_process/DestinationProcess.kt (hunk 5 above)
Comment (benmoriceau) [reply airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/test/util/destination_process/DestinationProcess.kt:65]: Done. (need to push)

**Pull request merged**